File: qqbot-plugin-spi/src/main/java/com/mieai/qqbot/plugin/spi/BotPluginFactory.java
package com.mieai.qqbot.plugin.spi;

import com.mieai.qqbot.plugin.api.PluginContext;

/** ServiceLoader entry point supplied by a trusted plugin JAR. */
public interface BotPluginFactory {
    String pluginId();
    BotPlugin create(PluginContext context);
}

File: qqbot-plugin-spi/src/main/kotlin/com/mieai/qqbot/plugin/spi/BotPluginFactory.kt
package com.mieai.qqbot.plugin.spi

import com.mieai.qqbot.plugin.api.PluginRuntimeContext

/** ServiceLoader entry point supplied by a trusted plugin JAR. */
interface BotPluginFactory {
    val pluginId: String

    fun create(context: PluginRuntimeContext): BotPlugin
}
